org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/editor/WRecordIDEditor.java

```diff
@@ -251,11 +262,16 @@ public void propertyChange(PropertyChangeEvent evt)
 		if (GridField.PROPERTY.equals(evt.getPropertyName()))
 		{
 			int tableID =  tableIDValue != null ? Integer.parseInt(String.valueOf(tableIDValue)) : 0;
-			int recordID = Integer.parseInt(Objects.toString(evt.getNewValue(), "-1"));
-			if (tableID > 0 && recordID >= 0)
-			{
-				MTable table = MTable.get(Env.getCtx(), tableID);
-				table.getPO(recordID, null);	// calls po.checkCrossTenant() method
+			MTable table = MTable.get(tableID);
```

Comment:
if tableID can be 0 here, I think it is better to wrap this block within if (tableID > 0) condition.

Reply:
done, thanks